presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -362,7 +377,44 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     conf,
                     typeManager);
         }
-        return new HiveWriter(hiveFileWriter, partitionName, isNew, fileNameWithExtension, write.toString(), target.toString());
+
+        String writerImplementation = hiveFileWriter.getClass().getName();
+
+        // TODO: the lambda takes an ugly parameter
+        Consumer<HiveWriter> onCommit = (HiveWriter hiveWriter) ->
+        {
+            long size;
+            try {
+                size = hdfsEnvironment.getFileSystem(session.getUser(), path, conf).getFileStatus(path).getLen();
+            }
+            catch (IOException e) {
+                // Do not fail the query if file system is not available
+                size = -1;
+            }
+
+            Optional<Principal> principal = session.getIdentity().getPrincipal();
+            String principalName = principal.isPresent() ? principal.get().getName() : "EMPTY";
+
+            WriteCompletedEvent event = new WriteCompletedEvent(
+                    session.getQueryId(),
+                    path.toString(),
+                    schemaName,
+                    tableName,
+                    partitionName.isPresent() ? partitionName.get() : "EMPTY",
+                    outputStorageFormat.getOutputFormat(),
+                    writerImplementation,
+                    nodeManager.getCurrentNode().getVersion(),
+                    nodeManager.getCurrentNode().getHttpUri().getHost(),
+                    principalName,
+                    nodeManager.getEnvironment(),
+                    null,   // TODO: session properties are not accessible through the interface; we might either (1) extent the interface, or (2) get the properties only we are interested in
```

Comment:
We should add a method to get all properties (for this connector).  Let's do this in another PR.